parser.go

```diff
@@ -138,3 +148,83 @@ func (p *Parser[V]) Pop() *Node[V] {
 	p.node = p.node.Parent
 	return n
 }
+
+// RotateLeft moves the current node to the position it's parent had,
+// making the parent into the current node's child.
+func (p *Parser[V]) RotateLeft() *Node[V] {
+	// n = node , op = original parent , gp = grand parent
+	n := p.node
+	op := n.Parent
+	gp := op.Parent
+
+	// Remove n from op's Children
+	opChildren := op.Children[:0]
+	for _, x := range op.Children {
```

Comment:
NOTE: We could eventually optimize this like `DeleteFunc` but it's fine as is for now.
https://cs.opensource.google/go/go/+/refs/tags/go1.21.6:src/slices/slices.go;l=230